src/main/java/io/supertokens/webserver/api/bulkimport/GetBulkImportUsers.java

```diff
@@ -0,0 +1,104 @@
+ *    obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *    License for the specific language governing permissions and limitations
+ *    under the License.
+ */
+
+package io.supertokens.webserver.api.bulkimport;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
+import io.supertokens.Main;
+import io.supertokens.bulkimport.BulkImport;
+import io.supertokens.bulkimport.BulkImportUserPaginationContainer;
+import io.supertokens.bulkimport.BulkImportUserPaginationToken;
+import io.supertokens.output.Logging;
+import io.supertokens.pluginInterface.exceptions.StorageQueryException;
+import io.supertokens.pluginInterface.multitenancy.AppIdentifierWithStorage;
+import io.supertokens.pluginInterface.multitenancy.exceptions.TenantOrAppNotFoundException;
+import io.supertokens.utils.Utils;
+import io.supertokens.webserver.InputParser;
+import io.supertokens.webserver.WebserverAPI;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+public class GetBulkImportUsers extends WebserverAPI {
+    public GetBulkImportUsers(Main main) {
+        super(main, "");
+    }
+
+    @Override
+    public String getPath() {
+        return "/bulk-import/users";
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        String status = InputParser.getQueryParamOrThrowError(req, "status", true);
+        String paginationToken = InputParser.getQueryParamOrThrowError(req, "paginationToken", true);
+        Integer limit = InputParser.getIntQueryParamOrThrowError(req, "limit", true);
+
+        if (limit != null) {
+            if (limit > BulkImport.GET_USERS_PAGINATION_LIMIT) {
+                throw new ServletException(
+                        new BadRequestException("Max limit allowed is " + BulkImport.GET_USERS_PAGINATION_LIMIT));
+            } else if (limit < 1) {
+                throw new ServletException(new BadRequestException("limit must a positive integer with min value 1"));
+            }
+        } else {
+            limit = BulkImport.GET_USERS_DEFAULT_LIMIT;
+        }
+
+        if (status != null
+                && !Arrays.asList("NEW", "PROCESSING", "FAILED").contains(status)) {
+            throw new ServletException(new BadRequestException(
+                    "Invalid value for status. Pass one of NEW, PROCESSING or, FAILED!"));
+        }
+
+        AppIdentifierWithStorage appIdentifierWithStorage = null;
+
+        try {
+            appIdentifierWithStorage = this.getAppIdentifierWithStorage(req);
```

Comment:
use getAppIdentifierWithStorageFromRequestAndEnforcePublicTenant instead (see similar comment in other api)

Reply:
fixed